inspirehep/modules/workflows/tasks/magpie.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Set of workflow tasks for MagPie API."""
+
+from __future__ import absolute_import, print_function
+
+from flask import current_app
+
+from inspirehep.modules.workflows.utils import json_api_request
+from inspirehep.utils.record import get_value
+
+
+def get_magpie_url():
+    """Return the Magpie URL endpoint, if any."""
+    base_url = current_app.config.get("MAGPIE_API_URL")
+    if not base_url:
+        return
+    return "{base_url}/predict".format(
+        base_url=base_url
+    )
+
+
+def prepare_magpie_payload(record, corpus):
+    """Prepare payload to send to Magpie API."""
+    payload = dict(text="", corpus="keywords")
+    titles = filter(None, get_value(record, "titles.title", []))
+    abstracts = filter(None, get_value(record, "abstracts.value", []))
+    payload["text"] = ". ".join(
+        titles.encode('utf-8') + abstracts.encode('utf-8')
+    )
+    return payload
+
+
+def filter_magpie_response(labels, limit):
+    """Filter response from Magpie API, keeping most relevant labels."""
+    filtered_labels = []
+    for word, score in labels:
+        if score >= limit:
+            filtered_labels.append((word, score))
```

Comment:
Yes, we can make this a one line comprehension as it should be readable enough. Thanks!